network/netcore/src/compat.rs

```diff
@@ -0,0 +1,87 @@
+use pin_project::pin_project;
```

Comment:
Can you add comments on what the `Compat` struct is useful for?

Reply:
Added some brief comments. I'm hoping that the community will end up standardizing on a set of traits but in the mean time we can use a compat layer.